packages/builder/src/components/backend/DataTable/RelationshipDataTable.svelte

```diff
@@ -0,0 +1,40 @@
+<script>
+  import api from "builderStore/api"
+  import Table from "./Table.svelte"
+  import { onMount } from "svelte"
+  import { backendUiStore } from "builderStore"
+
+  export let modelId
+  export let recordId
+  export let fieldName
+
+  let record
+  let title
+
+  $: data = record?.[fieldName] ?? []
+  $: linkedModelId = data?.length ? data[0].modelId : null
+  $: linkedModel = $backendUiStore.models.find(
+    model => model._id === linkedModelId
+  )
+  $: schema = linkedModel?.schema
+  $: model = $backendUiStore.models.find(model => model._id === modelId)
+  $: fetchData(modelId, recordId)
```

Comment:
I've seen a few issues in the past with async reactive statements doing a ton of unnecessary requests - worth checking out the network tab just to make sure.

Reply:
Cheers for the heads up. I can see how that would happen actually if you were referencing other variables in the function itself, but in this case I've made sure the fetch function is pure and only depends on those two parameters which are both primitives so will work with shallow equality checking (and verified via network tab) so it looks alright.